client/layout/masterbar/logged-out.jsx

```diff
@@ -64,6 +64,10 @@ class MasterbarLoggedOut extends PureComponent {
 			redirectTo,
 		} );
 
+		if ( currentQuery && currentQuery.partner_id ) {
+			loginUrl = addQueryArgs( { partner_id: currentQuery.partner_id }, loginUrl );
```

Comment:
Would be great if this can be done inside `login()`, as we currently handle several other `addQueryArgs` cases there. 

FWIW this works just fine as-is, and we'd probably have to inject the current query args into `login()` to make it work, so probably just go ahead and keep this as-is.